quora-api/src/main/java/com/upgrad/quora/api/exception/RestExceptionHandler.java

```diff
@@ -22,4 +23,17 @@
                 new ErrorResponse().code(exe.getCode()).message(exe.getErrorMessage()), HttpStatus.CONFLICT
         );
     }
+
+    /**
+     * This method handles the exception for failed Authentication
+     * @param exe
+     * @param request
+     * @return
+     */
+    @ExceptionHandler(AuthenticationFailedException.class)
+    public ResponseEntity<ErrorResponse> signinException(AuthenticationFailedException exe, WebRequest request) {
+        return new ResponseEntity<ErrorResponse>(
```

Comment:
The more correct Http status should be UNAUTHORIZED

Reply:
Changes made.